lib/Conversion/CalyxToHW/CalyxToHW.cpp

```diff
@@ -289,9 +289,9 @@ struct ConvertCellOp : public OpInterfaceConversionPattern<CellInterface> {
         .Case([&](NotLibOp op) {
           auto in =
               wireIn(op.getIn(), op.instanceName(), op.portName(op.getIn()), b);
-          auto one = b.create<hw::ConstantOp>(op.getIn().getType(), 0);
+          auto allOnes = b.create<hw::ConstantOp>(op.getIn().getType(), -1);
 
-          auto xorOp = b.create<XorOp>(in, one, false);
+          auto xorOp = b.create<XorOp>(in, allOnes, false);
```

Comment:
(I'm not familiar with CalyX but you should be able to use comb::createOrFoldNot for this
https://github.com/llvm/circt/blob/5061640e305e2415817e3cf59f682e16a6e0c246/lib/Dialect/Comb/CombOps.cpp#L53

```suggestion
          auto xorOp = comb::createOrFoldNot(in, b);
```

Reply:
Thanks, good idea!